### PR TITLE
Default Google Drive authentication to Secret Instead of Saved Credentials

### PR DESCRIPTION
**Pull request by sureshc**
Default our `Google::Drive` wrapper Class to authenticate to Google using our credentials stored in a Secret and populated in a `CDO` configuration setting instead of conditionally falling back to using session credentials saved on the local file system. Any feature using `Google::Drive` can chose to provide a different (non-default) set of authentication credentials.

We rebuilt the staging server and `bin/cron/import_google_sheets` stopped working because it was relying on credentials that may have been manually saved on the old server. Discussion [here](https://codedotorg.slack.com/archives/C03CK49G9/p1695840531522929).

## Testing story

I haven’t yet updated the tests to account for the new authentication strategy.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: bin/i18n/utils/malformed_i18n_reporter.rb [hunk 1]
```diff
@@ -51,7 +51,7 @@ def report
       def google_drive
         return @google_drive if defined? @google_drive
 
-        @google_drive = CDO.gdrive_export_secret && Google::Drive.new(service_account_key: CDO.gdrive_export_secret)
+        @google_drive = CDO.gdrive_export_secret && Google::Drive.new
```
Comment (Hamms) [inline bin/i18n/utils/malformed_i18n_reporter.rb]: tiny nit: I'd love it if this method didn't even need to know the exact name of the key any more; what do you think about adding a method on the `Google::Drive` class? Something like,

```ruby
def self.service_account_key
  StringIO.new(CDO.gdrive_export_secret&.to_json)
end
```

We could use it in the initialize method to make it even simpler than this PR already makes it, and use it here like

```suggestion
        @google_drive = Google::Drive.service_account_key.present? && Google::Drive.new
```

**Review by Hamms — APPROVED** [review]
One minor encapsulation tweak I'd like to explore, but this sounds like a good plan to me! Have we been able to verify whether staging is able to correctly import gsheets with this change?

**Comment by sureshc** [comment]
> Have we been able to verify whether staging is able to correctly import gsheets with this change?

I’ll checkout the file `lib/cdo/google/drive.rb` from this branch directly to staging and then monitor the import_google_sheets job to ensure it runs successfully

File: bin/i18n/utils/malformed_i18n_reporter.rb (hunk 1 above)
Comment (sureshc) [reply bin/i18n/utils/malformed_i18n_reporter.rb]: I’m wary of tinkering with this i18n stuff, but it seems to me like the check for the Secret is unnecessary. This feature is going to raise an error if the system running this i18n code doesn’t have access to the Secret. It’s just a question of whether that happens during instantiation of `Google::Drive` or a later attempt to call methods on `@google_drive`.

Comment (Hamms) [reply bin/i18n/utils/malformed_i18n_reporter.rb]: Lol, yeah, good point

**Review by Hamms — APPROVED** [review]

**Pull request merged**